packages/block-library/src/navigation/edit/index.js

```diff
@@ -224,36 +224,54 @@ function Navigation( {
 	// than 1 exists then use the most recent.
 	// The aim is for the block to "just work" from a user perspective using existing data.
 	useEffect( () => {
-		if (
-			hasUncontrolledInnerBlocks ||
-			isCreatingNavigationMenu ||
-			ref ||
-			! navigationMenus?.length
-		) {
+		// Only autofallback to published menus.
+		const fallbackNavigationMenus = navigationMenus?.filter(
+			( menu ) => menu.status === 'publish'
+		);
+
+		if ( hasUncontrolledInnerBlocks || isCreatingNavigationMenu || ref ) {
 			return;
 		}
 
-		navigationMenus.sort( ( menuA, menuB ) => {
-			const menuADate = new Date( menuA.date );
-			const menuBDate = new Date( menuB.date );
-			return menuADate.getTime() < menuBDate.getTime();
-		} );
+		// See if we have any navigation menus
+		if ( fallbackNavigationMenus?.length ) {
+			navigationMenus.sort( ( menuA, menuB ) => {
+				const menuADate = new Date( menuA.date );
+				const menuBDate = new Date( menuB.date );
+				return menuADate.getTime() < menuBDate.getTime();
+			} );
 
-		// Only autofallback to published menus.
-		const fallbackNavigationMenus = navigationMenus.filter(
-			( menu ) => menu.status === 'publish'
-		);
-		if ( fallbackNavigationMenus.length === 0 ) return;
+			/**
+			 *  This fallback displays (both in editor and on front)
+			 *  a list of pages only if no menu (user assigned or
+			 *  automatically picked) is available.
+			 *  The fallback should not request a save (entity dirty state)
+			 *  nor to be undoable, hence why it is marked as non persistent
+			 */
+			__unstableMarkNextChangeAsNotPersistent();
+			setRef( fallbackNavigationMenus[ 0 ].id );
+		} else if ( classicMenus?.length ) {
+			// Use a classic menu fallback
 
-		/**
-		 *  This fallback displays (both in editor and on front)
-		 *  a list of pages only if no menu (user assigned or
-		 *  automatically picked) is available.
-		 *  The fallback should not request a save (entity dirty state)
-		 *  nor to be undoable, hence why it is marked as non persistent
-		 */
-		__unstableMarkNextChangeAsNotPersistent();
-		setRef( fallbackNavigationMenus[ 0 ].id );
+			// We should get the primary one.
+
+			// We should probably create this not as a draft...
+
+			// This is duplicated several times.
```

Comment:
Here we are going back to the problem of converting a classic menu to a block nav menu automatically on page load, if we do it as a draft, we will have duplicated draft menus. If we do it as published and the user reloads without saving, classic menu will be selected and shown in both editor and frontend.

Could be possible to add a nav menu from the classic menu on the editor without creating an entity in the database? Should we wait for slug references to add them as a check for not duplicating them?

Reply:
> Could be possible to add a nav menu from the classic menu on the editor without creating an entity in the database? Should we wait for slug references to add them as a check for not duplicating them?

The blocks could be 'uncontrolled' inner blocks (so I think the code would get the converted blocks from the classic menu and set them using `replaceBlocks`), but they're not the smoothest experience either. 

The nice thing about creating an entity properly is that the same fallback logic won't need to be added in the PHP render code, so it's probably best either not to create it as a draft, or do the slug reference checking thing.

Reply:
I guess the other option is that the creation of the wp_navigation happens server side, and then the client side fallback code that selects navigation menus would already handle the job on the client side.

There is a `gutenberg_convert_menu_items_to_blocks` PHP function already, but I don't know if it works perfectly. At a glance it doesn't look like it was ever updated to handle the navigation-submenu block.